lib/groupify/adapter/active_record/group.rb

```diff
@@ -50,7 +44,10 @@ def merge!(source)
 
       module ClassMethods
         def with_member(member)
-          member.groups
+          #member.groups
+          joins(:group_memberships_as_group).
+          merge(Groupify.group_membership_klass.where(member_id: member.id, member_type: member.class.base_class.to_s)).
```

Comment:
@dwbutler I refactored the queries to use `merge` instead of `where`, as you started doing. Tests still pass. As you indicated, there are probably very specific STI-related scenarios we need to create some tests for. I'll see what I can come up with as I use this in our project, or if this use of `merge` handles things for us.